XExtensionItem/XExtensionItem.m

```diff
@@ -193,7 +193,22 @@ - (id)activityViewController:(UIActivityViewController *)activityViewController
         item.attachments = ({
             id activityItem = [self activityItemForActivityType:activityType];
             
-            NSItemProvider *mainAttachment = [[NSItemProvider alloc] initWithItem:activityItem typeIdentifier:self.typeIdentifier];
+            NSString *typeIdentifier = ^NSString *{
+                BOOL classHasChanged = ![[activityItem class] isSubclassOfClass:[self.placeholderItem class]];
```

Comment:
Nope, I don’t feel strongly about this being simpler I just wanted to get a better understanding of what the complexity was buying us.

That said, I am now considering maybe waiting for 1.1 to include this? It doesn't seem like something that has to go into 1.0 and may be worth giving ourselves some extra time to think on the approach, solicit additional feedback, and add some test coverage.